lib/util/exec/exec-error.ts

```diff
@@ -0,0 +1,45 @@
+import type { RawExecOptions } from './types';
+
+export interface ExecErrorData {
+  cmd: string;
+  stderr: string;
+  stdout: string;
+  options: RawExecOptions;
+  exitCode?: number;
+  signal?: NodeJS.Signals;
+}
+
+export class ExecError extends Error {
+  cmd: string;
+  stderr: string;
+  stdout: string;
+  options: RawExecOptions;
+  exitCode?: number;
+  signal?: NodeJS.Signals;
+
+  constructor(message: string, data: ExecErrorData, err?: Error) {
+    const { cmd, exitCode, stderr, stdout, options, signal } = data;
+
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause
+    if (err) {
+      super(message, { cause: err });
+    } else {
+      super(message);
+    }
+
+    // Set the prototype explicitly: https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
+    Object.setPrototypeOf(this, ExecError.prototype);
```

Comment:
i don't think this is necessary. it's only relevant when downleveling typescript to es5, which we don't do.

Reply:
>This is due to the fact that constructor functions for Error, Array, and the like use ECMAScript 6's `new.target` to adjust the prototype chain; however, there is no way to ensure a value for `new.target` when invoking a constructor in ECMAScript 5

Thats correct, will be removed as we target es2020.

Reply:
i've also compared the output when running with and without this via a driver function, and both are basically the same.